presto-main/src/main/java/com/facebook/presto/execution/scheduler/SqlQueryScheduler.java

```diff
@@ -321,11 +226,10 @@ private void schedule()
                     break;
                 }
 
-                List<List<StageExecutionAndScheduler>> sectionStageExecutions = getStageExecutions(sectionsReadyForExecution);
-                sectionStageExecutions.forEach(scheduledStageExecutions::addAll);
-                sectionStageExecutions.stream()
-                        .map(executionInfos -> executionInfos.stream()
-                                .collect(toImmutableList()))
+                List<SectionExecution> sectionExecutions = getOrCreateStageExecutions(sectionsReadyForExecution);
```

Comment:
line 210:
```
 // This list is tracked to make sure everything can be closed
```

line 215: rename `sectionExecutionSchedules` to `executionSchedules`

Reply:
Per discussion, section can only be ready once. So the method name should just be `createStageExecutions` and throws when tries to re-create the same stage execution.